### ISSUE TITLE
importlib_metadata required but missing for Python 3.7

### ISSUE THREAD

**Issue by anergictcell** [pydantic/pydantic#7903]
### Initial Checks

- [X] I confirm that I'm using Pydantic V2

### Description

`pydantic==2.4.2` requires additional dependencies to work with `Python 3.7`

The file `pydantic/plugin/_loader.py` tries to import `importlib_metadata` for Python version < 3.8, but the backport library is not part of the dependencies of Pydantic2.

**Steps to reproduce:**

Setup clean new environment:

```bash
docker run --rm -it python:3.7 bash
```

Within the docker container:
```bash
pip install pydantic

cat > test.py << 'EOF'
from pydantic import BaseModel

class Foo(BaseModel):
    bar: int

def main(a: int) -> Foo:
    return Foo(bar=a)
EOF

python test.py
```

**Expected:**
The script runs successfully

**Observed:**
```
Traceback (most recent call last):
  File "test.py", line 3, in <module>
    class Foo(BaseModel):
  File "/usr/local/lib/python3.7/site-packages/pydantic/_internal/_model_construction.py", line 189, in __new__
    types_namespace=types_namespace,
  File "/usr/local/lib/python3.7/site-packages/pydantic/_internal/_model_construction.py", line 499, in complete_model_class
    cls.__pydantic_validator__ = create_schema_validator(schema, core_config, config_wrapper.plugin_settings)
  File "/usr/local/lib/python3.7/site-packages/pydantic/plugin/_schema_validator.py", line 28, in create_schema_validator
    from ._loader import get_plugins
  File "/usr/local/lib/python3.7/site-packages/pydantic/plugin/_loader.py", line 12, in <module>
    import importlib_metadata
ModuleNotFoundError: No module named 'importlib_metadata'
```

**Temporary fix**:
Install `importlib_metadata` manually

```bash
pip install importlib_metadata
```

**Proposed fix**
Update the `pyproject.toml` dependency section with
```toml
dependencies = [
    'typing-extensions>=4.6.1',
    'annotated-types>=0.4.0',
    "pydantic-core==2.11.0",
+  "importlib_metadata; python_version<='3.7'",
]
```

### Example Code

```Python
from pydantic import BaseModel

class Foo(BaseModel):
    bar: int

def main(a: int) -> Foo:
    return Foo(bar=a)
```

### Python, Pydantic & OS Version

```Text
pydantic version: 2.4.2
        pydantic-core version: 2.10.1
          pydantic-core build: profile=release pgo=false
                 install path: /usr/local/lib/python3.7/site-packages/pydantic
               python version: 3.7.12 (default, Nov 18 2021, 00:44:17)  [GCC 10.2.1 20210110]
                     platform: Linux-5.15.49-linuxkit-aarch64-with-debian-11.1
             related packages: typing_extensions-4.7.1
```

**Comment by sydney-runkle** [comment]
Hi @anergictcell,

Thanks for reporting this. Looks like a duplicate of #7792, but given the easy suggested fix, I'd be happy to fix this. Thanks!

**Comment by sydney-runkle** [comment]
Here's an example of where we use this and why it only needs to be fixed for 3.7:

https://github.com/pydantic/pydantic/blob/47aa97e5311ed40dd1fe70f6cade874b4dd429e5/pydantic/version.py#L24-L29
